tests/unit_tests/test_suassystem_sda_converter.py

```diff
@@ -13,21 +13,110 @@ def setUp(self):
+        expected_result=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_result[0], returned_array[0][0][0])
+        self.assertEqual(expected_result[1], returned_array[0][0][1])
+
+    def test_convert_fly_zone(self):
+        result_fly_zone=self.sda_converter.convert_fly_zone(self.boundary_points)
+        test_location1=Location(self.boundary_points[0]["boundary_pts"][0]["latitude"],self.boundary_points[0]["boundary_pts"][0]["longitude"],0)
+        expected_fly_zone=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_fly_zone[0], result_fly_zone[0][0])
+        self.assertEqual(expected_fly_zone[1], result_fly_zone[0][1])
+
+    def test_does_guided_path_exist(self):
+        self.sda_converter.set_guided_path(numpy.array([numpy.array([1,2,3]),numpy.array([2,2,3])]))
+        self.assertTrue(self.sda_converter.does_guided_path_exist())
+       
+    def test_has_path_changed(self):
+        path1=numpy.array([numpy.array([1,2,3]), numpy.array([2,2,3])])
+        path2=numpy.array([numpy.array([1,2,3]), numpy.array([2,2,3]),numpy.array([30,2,3])])
+        path3=numpy.array([numpy.array([1,2,3]), numpy.array([2,2,3]),numpy.array([4,2,4])])
+        self.assertTrue(self.sda_converter.has_path_changed(path1,path2))
+        self.assertTrue(self.sda_converter.has_path_changed(path2,path3))
+
+    def test_has_uav_completed_guided_path(self):
+        self.sda_converter.set_path_index(3)
+        self.assertTrue(self.sda_converter.has_uav_completed_guided_path())
+
+    def test_get_uav_avoid_coordinates(self):
+        self.sda_converter.obstacle_map.set_drone_position(numpy.array([0,0,0]))
+        self.sda_converter.set_guided_path([numpy.array([45,45,50]),numpy.array([90,90,50])])
+        actual_location = Location(38.870550801732293,-77.321086622253759,15.2399995123)
+        test_location = self.sda_converter.get_uav_avoid_coordinates()
+        self.assertTrue(abs(actual_location.get_lat() - test_location.get_lat()) < 0.0001)
+        self.assertTrue(abs(actual_location.get_lon() - test_location.get_lon()) < 0.0001)
+        self.assertTrue(abs(actual_location.get_alt() - test_location.get_alt()) < 0.0001)        
+
+    def test_get_distance_to_current_guided_waypoint(self):
+        self.sda_converter.obstacle_map.set_drone_position(numpy.array([0,0,0]))
+        self.sda_converter.set_guided_path(numpy.array([numpy.array([45,45,50]),numpy.array([90,90,50])]))
+        self.assertEqual(136.74794331177344,self.sda_converter.get_distance_to_current_guided_waypoint())
+
+    def test_has_uav_reached_guided_waypoint(self):
+        self.sda_converter.set_guided_path(numpy.array([numpy.array([45,45,50]),numpy.array([90,90,50])]))
+        self.assertEqual(False,self.sda_converter.has_uav_reached_guided_waypoint())
+        
     def test_set_waypoints(self):
 
         #Test the set_waypoints() method
```

Comment:
Generally, any public method, whether it be a class or function, should have multi-line docstring comments, like so:

```
"""Return a foobang

Optional plotz says to frobnicate the bizbaz first.
"""
```

[This is according to PEP8 standards.](https://www.python.org/dev/peps/pep-0008/#documentation-strings)

However, since you're doing unit tests, you don't need docstrings on any of the tests. But, if you do want to explain a function, you should use docstrings.

Also, this comment doesn't add any explanation to the function. **You need to delete this line.**